File: OKIS4Lab/Python/MainApp.py
class MainApp:

    def add(self, a, b):
        return a + b

    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        if b == 0:
            raise ZeroDivisionError("Division by zero")
        return a / b

    def is_even(self, n):
        return n % 2 == 0

    def factorial(self, n):
        if n < 0:
            raise ValueError("Negative number")
        f = 1
        for i in range(1, n + 1):
            f *= i
        return f

    def reverse_string(self, s):
        return s[::-1]

    def is_prime(self, n):
        if n < 2:
            return False
        for i in range(2, int(n**0.5) + 1):
            if n % i == 0:
                return False
        return True

File: OKIS4Lab/Java/src/main/java/org/example/MainApp.java
package org.example;

public class MainApp {

    public int add(int a, int b) { return a + b; }
    public int subtract(int a, int b) { return a - b; }
    public int multiply(int a, int b) { return a * b; }
    public double divide(int a, int b) {
    if(b == 0) throw new ArithmeticException("Division by zero");
    return a / (double) b;
    }
    public boolean isEven(int n) { return n % 2 == 0; }
    public int factorial(int n) {
    if(n < 0) throw new IllegalArgumentException("Negative number");
    int f = 1;
    for (int i = 1; i <= n; i++) f *= i;
    return f;
    }
    public String reverseString(String s) {
        return new StringBuilder(s).reverse().toString();
    }
    public boolean isPrime(int n) {
        if(n < 2) return false;
        for(int i = 2; i*i <= n; i++)
            if(n % i == 0) return false;
        return true;
    }
}
